PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/cloudshell/controlchannel/UrlControlMessageHandler.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.plugins.azure.cloudshell.controlchannel
+
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import com.intellij.ide.BrowserUtil
+import com.intellij.openapi.diagnostic.Logger
+
+class UrlControlMessageHandler(
+        private val gson: Gson)
+    : ControlMessageHandler {
+
+    companion object {
+        private val logger = Logger.getInstance(UrlControlMessageHandler::class.java)
+    }
+
+    override fun handle(jsonControlMessage: String) {
+        val message = gson.fromJson(jsonControlMessage, UrlControlMessage::class.java)
+
+        if (!message.url.isEmpty()) {
+            logger.info("Opening browser for URL: {message.url}")
+
+            // Azure needs some time to get ready to serve the URL, otherwise we will open a 404/500 error page
+            Thread.sleep(2000)
```

Comment:
Can we do check for response status before open and wait till we get 200 response. Also, can we move this to background able task to make sure user understands what's going on?

Reply:
Updated, introduced a helper function for this. Commit: https://github.com/maartenba/azure-tools-for-intellij/commit/145111c6eb9400d0528c618ebc282ef0ddb7289a